pkg/jx/cmd/step_create_task.go

```diff
@@ -167,89 +170,103 @@ func (o *StepCreateTaskOptions) Run() error {
 			return util.MissingOption("docker-registry")
 		}
 	}
-	if o.BuildPackURL == "" || o.BuildPackRef == "" {
-		if o.BuildPackURL == "" {
-			o.BuildPackURL = settings.BuildPackURL
-		}
-		if o.BuildPackRef == "" {
-			o.BuildPackRef = settings.BuildPackRef
-		}
-	}
-	if o.BuildPackURL == "" {
-		return util.MissingOption("url")
-	}
-	if o.BuildPackRef == "" {
-		return util.MissingOption("ref")
-	}
-	if o.PipelineKind == "" {
-		return util.MissingOption("kind")
-	}
+
 	if o.Dir == "" {
 		o.Dir, err = os.Getwd()
 		if err != nil {
 			return err
 		}
 	}
-	projectConfig, projectConfigFile, err := config.LoadProjectConfig(o.Dir)
-	if err != nil {
-		return errors.Wrapf(err, "failed to load project config in dir %s", o.Dir)
-	}
-	if o.Pack == "" {
-		o.Pack = projectConfig.BuildPack
-	}
-	if o.Pack == "" {
-		o.Pack, err = o.discoverBuildPack(o.Dir, projectConfig)
-	}
 
-	if o.Pack == "" {
-		return util.MissingOption("pack")
-	}
+	// TODO: Best to separate things cleanly into 2 steps: creation of CRDs and
+	// application of those CRDs to the cluster. Step 2 should be identical both
+	// cases, so we'd just need a flag to switch the single function that is used
+	// to generate stuff and then everything else would be identical.
+	if !o.FromYaml {
+		if o.BuildPackURL == "" || o.BuildPackRef == "" {
+			if o.BuildPackURL == "" {
+				o.BuildPackURL = settings.BuildPackURL
+			}
+			if o.BuildPackRef == "" {
+				o.BuildPackRef = settings.BuildPackRef
+			}
+		}
+		if o.BuildPackURL == "" {
+			return util.MissingOption("url")
+		}
+		if o.BuildPackRef == "" {
+			return util.MissingOption("ref")
+		}
+		if o.PipelineKind == "" {
+			return util.MissingOption("kind")
+		}
+		projectConfig, projectConfigFile, err := config.LoadProjectConfig(o.Dir)
```

Comment:
Actually, I think this may be what we need - this is looking for `jenkins-x.yml` in the repo already, and I'm planning to mesh our syntax into that. So then we could theoretically pull the whole Pipeline from `projectConfig` if it's present, and otherwise fall back on `BuildPack`.